overlord/configstate/configcore/watchdog.go

```diff
@@ -0,0 +1,109 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+var watchdogConfigKeys = map[string]bool{
+	"RuntimeWatchdogSec":  true,
+	"ShutdownWatchdogSec": true,
+}
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.watchdog.runtime"] = true
+	supportedConfigurations["core.watchdog.shutdown"] = true
+}
+
+func watchdogConfEnvironment() string {
+	return filepath.Join(dirs.GlobalRootDir, "/etc/systemd/system.conf.d/10-watchdog.conf")
+}
+
+func updateWatchdogConfig(path string, config map[string]string) error {
+	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return nil
+	}
+	defer f.Close()
+
+	content := "[Manager]\n"
+	for k, v := range config {
+		if len(v) > 0 {
+			content += fmt.Sprintf("%s=%s\n", k, v)
+		}
+	}
+
+	return ioutil.WriteFile(path, []byte(content), 0644)
```

Comment:
This is writing the file a second time, while the first time (f) is still open and will be closed afterwards.